### DISCUSSION TITLE
nomenclature 包不生成符号表

### DISCUSSION THREAD

**Discussion by skyzh** [sjtug/SJTUThesis#642] [提问] [answered]
在 SJTUThesis 模版中使用 nomenclature 时，不会生成符号表。

**Comment by skyzh** [comment, marked as answer]
nomenclature 需要生成一个索引文件。可以参考 https://tex.stackexchange.com/questions/310635/nomenclatures-not-working-when-using-package-nomencl 修改 latexmkrc 文件。

```
add_cus_dep('nlo', 'nls', 0, 'run_makenomencl');
push @generated_exts, 'nlo', 'nls';

sub run_makenomencl {
    if ( $silent ) {
        system "makeindex -q '$_[0].nlo' -s nomencl.ist -o '$_[0].nls'";
    }
    else {
        system "makeindex '$_[0].nlo' -s nomencl.ist -o '$_[0].nls'";
    };
}
```

<img width="488" alt="image" src="https://user-images.githubusercontent.com/4198311/119352299-1832c080-bcd4-11eb-91c2-86b376cf6c50.png">

**Reply by AlexaraWu** [reply]
https://ctan.org/tex-archive/support/latexmk/example_rcfiles

latexmkrc 的各种配置，供参考。
